### PR TITLE
Use of cbuf_t for shared memory identifiers

### PR DESCRIPTION
**Pull request by phanikishoreg**
### Summary of this Pull Request (PR)

Fixes issue #327 using cbuf_t as the data-type for shared memory identifiers instead of `int`. 

### Intent for your PR

Choose one (Mandatory):

- [ ] This PR is for a code-review and is intended to get feedback, but not to be pulled yet.
- [x] This PR is mature, and ready to be integrated into the repo.

### Reviewers (Mandatory):
@gparmer 

### Code Quality

As part of this pull request, I've considered the following:

[Style](https://github.com/gparmer/composite/raw/ppos/doc/style_guide/composite_coding_style.pdf):

- [x] Comments adhere to the Style Guide (SG)
- [x] Spacing adhere's to the SG
- [x] Naming adhere's to the SG
- [x] All other aspects of the SG are adhered to, or exceptions are justified in this pull request
- [ ] I have run the auto formatter on my code before submitting this PR (see doc/auto_formatter.md for instructions)

[Code Craftsmanship](http://www2.seas.gwu.edu/~gparmer/posts/2016-03-07-code-craftsmanship.html):

- [x] I've made an attempt to remove all redundant code
- [x] I've considered ways in which my changes might impact existing code, and cleaned it up
- [x] I've formatted the code in an effort to make it easier to read (proper error handling, function use, etc...)
- [x] I've commented appropriately where code is tricky
- [x] I agree that there is no "throw-away" code, and that code in this PR is of high quality

### Testing

I've tested the code using the following test programs (provide list here):

- microbooter, capmgr unit tests.

### CODE REVIEW

File: src/components/implementation/capmgr/naive/cap_info.c [hunk 1]
```diff
@@ -193,64 +193,72 @@ __cap_cos_shared_page_allocn(struct cos_compinfo *rci, int num_pages, vaddr_t *c
...
-	if (__cap_info_shm_capmgr_vaddr(fidx) != 0) goto done;
-	if (rsh->shm_addr[fidx] != 0) goto done;
+	if (__cap_info_shm_capmgr_vaddr(fid)) goto done;
+	if (cap_shmem_region_vaddr(rsh, fid)) goto done;
 
-	rglb->region_npages[fidx] = num_pages;
-	ps_faa((long unsigned *)&(rglb->total_pages), num_pages);
-	ps_faa((long unsigned *)&(rsh->total_pages), num_pages);
+	rglb->region_npages[fid - 1] = num_pages;
+	ps_faa((unsigned long *)&(rglb->total_pages), num_pages);
```
Comment (gparmer) [inline src/components/implementation/capmgr/naive/cap_info.c]: wow, don't know how I missed the previous code.  Good fix.

File: src/components/implementation/capmgr/naive/mem_mgr.c [hunk 2]
```diff
@@ -25,41 +25,41 @@ memmgr_heap_page_allocn(unsigned int npages)
...
+	*pgaddr = cap_shmem_region_vaddr(cur_shi, shmid);
 
 done:
-	return shmidx;
+	return shmid;
 }
 
-int
-memmgr_shared_page_map_cserialized(vaddr_t *pgaddr, int *unused, int idx)
+u32_t
```
Comment (gparmer) [inline src/components/implementation/capmgr/naive/mem_mgr.c]: Little strange the `shared_mapn` still returns an `int`.  What do you think about this being a `size_t` instead?

I'm just having a little trouble understanding why `u32_t` is the right type for this.

File: src/components/implementation/capmgr/naive/cap_info.c [hunk 3]
```diff
@@ -179,7 +179,7 @@ __cap_cos_shared_page_mapn(struct cos_compinfo *rci, int num_pages, vaddr_t capv
 }
 
 static int
-__cap_cos_shared_page_allocn(struct cos_compinfo *rci, int num_pages, vaddr_t *capvaddr, vaddr_t *compvaddr)
+__cap_cos_shared_page_allocn(struct cos_compinfo *rci, u32_t num_pages, vaddr_t *capvaddr, vaddr_t *compvaddr)
```
[the remark this answers is not in the document]
Comment (gparmer) [reply src/components/implementation/capmgr/naive/cap_info.c]: This is a very good point.

I don't like `u32_t`, but you're likely right about `size_t`.  Maybe just unsigned long?

[the remark this answers is not in the document]
Comment (phanikishoreg) [reply src/components/implementation/capmgr/naive/cap_info.c]: Changed it and updated the PR with that commit! Thank you!

**Pull request merged**